### PR TITLE
add a new mode days-interval

### PR DESCRIPTION
**Pull request by sebdiem**
In this new mode of the date picker, the user can first select a date
and then select another date to form a date interval.
All the days in between are highlighted to provide visual feedback to
the user.

### CODE REVIEW

**Comment by sebdiem** [comment]
we're still below 500 LOC ;)
```
$ wc -l src/*.js
      20 src/dateutils.js
     400 src/index.js
      47 src/utils.js
     467 total
```

File: src/index.js [hunk 1]
```diff
@@ -250,9 +264,19 @@ export function init (
             'year': clickedDay.year(),
             'month': clickedDay.month(),
             'date': clickedDay.date(),
+            'hour': 0,
+            'minute': 0,
+            'second': 0,
```
Comment (vperron) [inline src/index.js]: j'ai peut que ça ne casse le comportement dans certains cas quand on peut avoir jusqu'à l'heure. Faudrait que je vérifie ou ajoute un test spécifique

File: src/index.js [hunk 2]
```diff
@@ -284,10 +308,37 @@ export function init (
...
+      mouseover: function (e) {
+        if (mode !== MODE_DAYS_INTERVAL || !selectedDateIntervalStart) {
+          return
+        }
+        const endIntervalDate = moment(e.target.dataset.value)
+        for (let el of Array.from(root.querySelectorAll('.' + styles.dow))) {
+          const currentDate = moment(el.dataset.value)
+          if (
+              (selectedDateIntervalStart <= currentDate && currentDate <= endIntervalDate) ||
+              (selectedDateIntervalStart >= currentDate && currentDate >= endIntervalDate)
```
Comment (vperron) [inline src/index.js]: je crois qu'ilvaut mieux utiliser `.isAfter` et `.isBefore` sinon il compare string à string.

File: src/index.js (hunk 1 above)
Comment (sebdiem) [reply src/index.js]: bien vu ! ça casse en effet

File: src/index.js (hunk 2 above)
Comment (sebdiem) [reply src/index.js]: cool deux fonctions de plus de l'API moment dans ma tête :)

**Comment by sebdiem** [comment]
@vperron I added some tests + a Travis-CI config (build passing on a fork of mine)
Could you please : 
- enable bluepicker on Travis CI (don't have the rights to do so)
- add more tests as you wish to "lock" the API of the component (I don't have all the subtleties in mind : in particular I am not sure whether the UTC switch is working as intended)

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const webpackConfig = require('./webpack.config.js')
+const karmaDefaults = require('systematic').karma_get_defaults(__dirname, webpackConfig)
+
+module.exports = (karma) => karma.set(karmaDefaults)
```
Comment (vperron) [inline karma.conf.js:4]: not sure this is necessary. Is it ? Maybe with an update of systematic to pickup the webpack config correctly ?

**Comment by vperron** [comment]
Fantastic job !!! Agreed. I'll add it to Travis ASAP

**Comment by vperron** [comment]
Can you please update the dependencies and close the PR ?

File: karma.conf.js (hunk 3 above)
Comment (sebdiem) [reply karma.conf.js:4]: yes it is dot files are not loaded correctly without it

**Pull request merged**